crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,437 @@
+        pub size_constraints: &'static SizeConstraints,
+        /// The margin, padding and border of the UI node
+        pub spacing: &'static Spacing,
+        /// The flexbox layout parameters
+        pub flex_layout: &'static FlexLayout,
+        /// The direction of the text
+        pub text_direction: &'static TextDirection,
+        /// The behavior in case the node overflows its allocated space
+        pub overflow: &'static Overflow,
+    }
+
+    /// A type alias for when any of the components in a [`FlexLayoutQuery`] have changed.
+    pub type FlexLayoutChanged = Or<(
+        Changed<LayoutStrategy>,
+        Changed<PositionType>,
+        Changed<SizeConstraints>,
+        Changed<Spacing>,
+        Changed<FlexLayout>,
+        Changed<TextDirection>,
+        Changed<Overflow>,
+    )>;
+
+    /// The flexbox-specific layout configuration of a UI node
+    ///
+    /// This follows the web spec closely,
+    /// you can use [guides](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) for additional documentation.
+    #[derive(Component, Serialize, Deserialize, Reflect, Debug, PartialEq, Clone, Copy)]
+    #[reflect_value(PartialEq, Serialize, Deserialize)]
+    pub struct FlexLayout {
+        /// How items are ordered inside a flexbox
+        ///
+        /// Sets the main and cross-axis: if this is a "row" variant, the main axis will be rows.
+        pub flex_direction: FlexDirection,
+        /// Aligns this container's contents according to the cross-axis
+        pub align_items: AlignItems,
+        /// Overrides the inherited [`AlignItems`] value for this node
+        pub align_self: AlignSelf,
+        /// Aligns this containers lines according to the cross-axis
+        pub align_content: AlignContent,
+        /// Aligns this containers items along the main-axis
+        pub justify_content: JustifyContent,
+        /// Controls how the content wraps
+        pub wrap: Wrap,
+        /// Defines how much a flexbox item should grow if there's space available
+        pub grow: f32,
+        /// How to shrink if there's not enough space available
+        pub shrink: f32,
+        /// The initial size of the item
+        pub basis: Val,
+    }
+
+    impl Default for FlexLayout {
+        fn default() -> FlexLayout {
+            FlexLayout {
+                flex_direction: Default::default(),
+                align_items: Default::default(),
+                align_self: Default::default(),
+                align_content: Default::default(),
+                justify_content: Default::default(),
+                wrap: Default::default(),
+                grow: 0.0,
+                shrink: 1.0,
+                basis: Val::Auto,
+            }
+        }
+    }
+
+    /// Defines how flexbox items are ordered within a flexbox
+    #[derive(Copy, Clone, PartialEq, Eq, Debug, Default, Serialize, Deserialize, Reflect)]
+    #[reflect_value(PartialEq, Serialize, Deserialize)]
+    pub enum FlexDirection {
```

Comment:
```suggestion
    pub enum Direction {
```
Rename to keep consistency with others members of `FlexLayout`, which doesn't prefix `Flex`

Reply:
@Weibye We should address this one.